tests/test_scylla.c

```diff
@@ -0,0 +1,177 @@
+      {TRANSACTION_5}   // approvees T7
+  };
+
+  hash243_queue_t transactions = NULL;
+  hash243_queue_entry_t* iter243 = NULL;
+  hash243_queue_t bundles = NULL;
+  hash243_queue_t addresses = NULL;
+  hash243_queue_t approvees = NULL;
+  hash243_queue_push(&bundles, (flex_trit_t const* const)BUNDLE_1);
+  hash243_queue_push(&bundles, (flex_trit_t const* const)BUNDLE_2);
+
+  hash243_queue_push(&addresses, (flex_trit_t const* const)ADDRESS_1);
+  hash243_queue_push(&addresses, (flex_trit_t const* const)ADDRESS_2);
+
+  hash243_queue_push(&approvees, (flex_trit_t const* const)TRANSACTION_6);
+  hash243_queue_push(&approvees, (flex_trit_t const* const)TRANSACTION_7);
+
+  get_transactions(session, &transactions, bundles, addresses, approvees);
+
+  size_t idx = 0;
+  CDL_FOREACH(transactions, iter243) {
+    TEST_ASSERT_FALSE(memcmp(iter243->hash, (flex_trit_t*)expected_response_transactions[idx++],
+                             sizeof(flex_trit_t) * NUM_FLEX_TRITS_HASH));
+  }
+  TEST_ASSERT_EQUAL_INT(idx, sizeof(expected_response_transactions) / (sizeof(char) * NUM_FLEX_TRITS_HASH));
+  hash243_queue_free(&bundles);
+  hash243_queue_free(&addresses);
+  hash243_queue_free(&approvees);
+  hash243_queue_free(&transactions);
+  return;
+}
+
+void test_scylla(void) {
+  /* input_transactions' graph
+   * T1-T2 in BUNDLE 1 approved T3-T4 and T5-T6
+   * T3-T4 in BUNDLE 2 approved T7-T8 and T9
+   * T5-T6 in BUNDLE 3 approved T7-T8 and T9
+   *
+   *    T1-T2
+   *    | /  \
+   *    |/    \
+   *    T3-T4  T5-T6
+   *    | /__\_/__/
+   *    |/    \  /
+   *    T7-T8   T9
+   */
+
+  status_t ret = SC_OK;
+  CassCluster* cluster = NULL;
+  CassSession* session = cass_session_new();
+  char hashes[][NUM_FLEX_TRITS_HASH] = {
+      {TRANSACTION_1}, {TRANSACTION_2}, {TRANSACTION_3}, {TRANSACTION_4}, {TRANSACTION_5}, {TRANSACTION_6},
+  };
+  char bundles[][NUM_FLEX_TRITS_BUNDLE] = {
+      {BUNDLE_1}, {BUNDLE_1}, {BUNDLE_2}, {BUNDLE_2}, {BUNDLE_3}, {BUNDLE_3},
+  };
+  char addresses[][NUM_FLEX_TRITS_ADDRESS] = {
+      {ADDRESS_1}, {ADDRESS_2}, {ADDRESS_3}, {ADDRESS_2}, {ADDRESS_4}, {ADDRESS_2},
+  };
+  char messages[][NUM_FLEX_TRITS_MESSAGE] = {
+      {MESSAGE_1}, {MESSAGE_2}, {MESSAGE_3}, {MESSAGE_4}, {MESSAGE_5}, {MESSAGE_6},
+  };
+  char timestamps[][NUM_FLEX_TRITS_TIMESTAMP] = {
+      {TIMESTAMP_1}, {TIMESTAMP_1}, {TIMESTAMP_2}, {TIMESTAMP_2}, {TIMESTAMP_3}, {TIMESTAMP_3},
+  };
+  char trunks[][NUM_FLEX_TRITS_TRUNK] = {
+      {TRANSACTION_2}, {TRANSACTION_3}, {TRANSACTION_4}, {TRANSACTION_7}, {TRANSACTION_6}, {TRANSACTION_7},
+  };
+  char branches[][NUM_FLEX_TRITS_BRANCH] = {
+      {TRANSACTION_3}, {TRANSACTION_5}, {TRANSACTION_7}, {TRANSACTION_9}, {TRANSACTION_7}, {TRANSACTION_9},
+  };
+  size_t tx_num = sizeof(hashes) / (NUM_FLEX_TRITS_HASH);
+  scylla_iota_transaction_t* transaction;
+
+  ret = init_scylla(&cluster, session, host, true);
+  TEST_ASSERT_EQUAL_INT(ret, SC_OK);
+  if (ret != SC_OK) {
```

Comment:
If `ret != SC_OK` here, I think the code would shudown without processing the next line

Reply:
Oh really!
Then I need to rework the flow of testing, or it will shudown without free and release.

Reply:
I am not sure is it a problem actually

Reply:
I think it is OK in modern OS
https://stackoverflow.com/questions/654754/what-really-happens-when-you-dont-free-after-malloc